hardware_integration/src/can_hardware_interface.cpp

```diff
@@ -372,77 +345,73 @@ void CANHardwareInterface::can_thread_function()
 
 	while (threadsStarted)
 	{
-		std::unique_lock<std::mutex> lMutex(threadMutex);
-		threadConditionVariable.wait(lMutex);
-		CanHardware *pCANHardware;
+		std::unique_lock<std::mutex> threadLock(threadMutex);
+		threadConditionVariable.wait(threadLock);
 
 		if (threadsStarted)
 		{
 			for (std::uint32_t i = 0; i < hardwareChannels.size(); i++)
```

Comment:
Can probably change to size_t here also - I guess it's just a general comment we might want to adjust those throughout while we're reworking this file